src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportDestinationClient/AzureExportDestinationClient.cs

```diff
@@ -0,0 +1,104 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Export.ExportDestinationClient
+{
+    public class AzureExportDestinationClient : IExportDestinationClient
+    {
+        private CloudBlobClient _blobClient = null;
+        private CloudBlobContainer _blobContainer = null;
+
+        private Dictionary<Uri, CloudBlockBlobWrapper> _uriToBlobMapping = new Dictionary<Uri, CloudBlockBlobWrapper>();
+        private Dictionary<(Uri FileUri, uint PartId), Stream> _streamMappings = new Dictionary<(Uri FileUri, uint PartId), Stream>();
+
+        public string DestinationType => "azure-block-blob";
+
+        public async Task ConnectAsync(string connectionSettings, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNullOrWhiteSpace(connectionSettings, nameof(connectionSettings));
+
+            CloudStorageAccount cloudAccount = null;
+            if (!CloudStorageAccount.TryParse(connectionSettings, out cloudAccount))
+            {
+                throw new CantConnectToDestinationException();
```

Comment:
Maybe just a generic `DestinationConnectionException` which can be used when connection settings is not valid or when the write operation fails?